backend/app/views.py

```diff
@@ -112,24 +114,129 @@ def search_photographers(request):
+    if location is not None and location.strip() != "" and location in locations:
+        search_params["approx_loc"] = location 
+    if map_square is not None and map_square.strip() != "":
+        try:
+            map_square = int(map_square)
+            if map_square in squares:
+                search_params["map_square"] = map_square 
+        except TypeError:
+            pass
+    
+    # Planning to check for multiple name starts for this field 
+    # Implmenetaiton example in this stackoverflow entry 
+    #  (https://stackoverflow.com/questions/5783588/django-filter-on-same-option-with-multiple-possibilities)
+    if name_start is not None and name_start.strip() != "" and name_start in nameStartsWith:
+        search_params["name__istartswith"] = name_start 
+    print(search_params)
+
+    order_by_field = parse_order_by(order_by)
+
+    if len(search_params) == 0:
+        matching_photographers = Photographer.objects.all()
     else:
-        matching_photographers = (Photographer.objects.all()
-                                                      .order_by("name")
-                                                      .prefetch_related("photo_set")[:50])
+        matching_photographers = Photographer.objects.filter(**search_params)
+
+    matching_photographers.prefetch_related("photo_set")
+
+    if order_by_field is not None:
+        matching_photographers = matching_photographers.order_by(order_by_field)
+    photographers_paginator = Paginator(matching_photographers, count_per_page)
+    current_page = photographers_paginator.get_page(page_number)
 
     serialized_photographers = (
-        PhotographerSearchSerializer(matching_photographers, many=True)
+        PhotographerSearchSerializer(current_page.object_list, many=True)
     ) # add pagination here
-    res = Response(serialized_photographers.data)
+    res = Response({
+        "page_number": page_number,
+        "results": serialized_photographers.data,
+        "is_last_page": not current_page.has_next()
+    })
     return res
 
 
+@api_view(['GET'])
+def get_search_photographers_dropdown_options(request):
+    """
+    API endpoint to get a list of photographers based on a search query that looks the photographers by name 
+    If not given a search query it will return the first 50 photographers sorted by name
+
+    TODO: Add pagination for both cases (when given a search query and when nothing is given) 
+    so that the user is sent the first 50 results and they can view more results as they scroll down the page.
+    """
+    locations = sorted(
+        filter(
+            lambda x: x is not None, 
+            list(
+                set(Photographer.objects.all().values_list('approx_loc', flat=True))
+            )
+        )
+    )
+
+    squares = sorted(
+        filter(
+            lambda x: x is not None, 
+            list(
+                set(Photographer.objects.all().values_list('map_square_id', flat=True))
```

Comment:
We should be sending map square numbers, not ids here. We shouldn't rely on the ID and map square number being the same.